config/opal_configure_options.m4

```diff
@@ -36,10 +36,9 @@ opal_show_subtitle "General configuration options"
 
 
 #
-# Is this a developer copy?
+# Is this a developer copy? Check for a known OMPI git hash
 #
-
-if test -d .git; then
+if git describe 43a3f4282055c7116ca618c17a9f27247f4923d2 &> /dev/null ; then
```

Comment:
I had to think about this a bit; I had a knee-jerk "eww!" reaction to this, but I can't quite quantify why.  I.e., it *feels* weird to do this, but I can't find any flaw in the reasoning behind it.

Can we make a few minor changes, though?

1. The code @rhc54 suggested looking for the `git` command seems reasonable; we don't want an error to occur here if there's no `git` command available.
2. Can you put in a comment explaining why we're doing this?  Some points to make, in no particular order:
    * We want to know if we're in a Git tree, and `git describe` will walk backwards up the tree to catch the case of VPATH builds in a subdirectory of a git clone
    * We know that this won't catch VPATH builds outside of a git clone.  It's not meant to be a perfect solution.
    * The intent of checking for a specific commit is to ensure that we're in an **OMPI** git clone (not just any arbitrary git clone) -- e.g., if someone expands an OMPI tarball in a git clone of some other repo (could happen in Homebrew, Spack, or EasyBuild, or some other build-scripts kind of git repo), we wouldn't want that tarball build to default to OMPI-developer values.
    * Instead of checking for 43a3f42 (which is a pretty recent commit), can you check for 350564b9f381dfbdbe119f26585f07da6f4b9e8a?  That's the root commit of OMPI's `main`.  I realize this won't make much of a functional difference compared to checking for 43a3f42, but it's a bit more of a definitive location to check for, and easy to describe ("we're looking for the known root commit of Open MPI's `main` repo").